content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx

```diff
@@ -1010,9 +994,72 @@ Support for [`autoincrement()`](../../../reference/api-reference/prisma-schema-r
 
 **Relational database connectors** implement `autoincrement()`, `dbgenerated()`, and `now()` at database level. The **MongoDB connector** does not support `autoincrement()`, and `now()` is implemented at Prisma level. `dbgenerated()` can only be used to generate an `ObjectId`.
 
-## Relations
+## Constraints and indexes in the Prisma schema
 
-Refer to the [relations documentation](relations) for more examples and information about relationships between models.
+Constraints and indexes are represented by [attributes](#defining-attributes) in the Prisma schema. For example:
+
+- `@unique` represents a unique constraint on a single column
+- `@@index` represets an index on one or more columns
+- `@relation` represents a foreign key constraint
+
+Constraints and indexes have **names** in the underlying database. You can [define constraint and index names in the Prisma schema in 2.27.0 and later](#constraint-and-index-names-preview). 
+
+Foreign key constraints - also called referential constraints - also define a referential **action**. You can [define referential actions in the Prisma schema in 2.26.0 and later](#referential-actions-preview).
+
+### Constraint and index names (Preview)
+
+In [2.27.0](https://github.com/prisma/prisma/releases/tag/2.27.0) and later, you can use the `map` field to define the **underlying constraint and index names** in the Prisma schema. The `map` field is _only_ visible in the schema if you use a constraint name that differs from Prisma's naming convention.
+
+Named constraints is a Preview feature. To enable this feature:
+
+1. Add `namedConstraints` to the `previewFeatures` field in the `generator` block:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["namedConstraints"]
+}
+```
+
+2. Choose an [an upgrade path](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2270#optional-enable-the-namedconstraints-preview-feature) depending on whether you use the Prisma Migrate or Introspection workflow.
+
+   <Admonition type="alert">
+
+   If you use Prisma Migrate and want to maintain your existing constraint and index names, **do not** enable the `namedConstraints` feature and immediately run `prisma migrate dev`. This will **change any underlying constraint name that does not follow Prisma's convention**. Follow the [upgrade path for the Prisma Migrate workflow](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2270#prisma-migrate-workflow).
+   
+   </Admonition>
+
+In the following schema defines two constraints that follow convention, and one constraint and an index that do not:
```

Comment:
A bit hard to follow. Maybe the same schema once with default, and once with customized instead?

Reply:
Expanded - now with constraints/indexes before and after custom names